docs/src/components/docs/SideNav.js

```diff
@@ -0,0 +1,153 @@
+import * as React from "react"
+import { sortBy, upperFirst, camelCase } from "lodash/fp"
+import { graphql, useStaticQuery } from "gatsby"
+import { Box, Heading } from "@chakra-ui/core"
+import { ComponentLink, TopNavLink } from "./NavLink"
+
+const sortByOrder = sortBy(["frontmatter.order"])
+const sortByTitle = sortBy(["frontmatter.title"])
+
+const useLinksQuery = () => {
+  return useStaticQuery(
+    graphql`
+      query LinksQuery {
+        main: allMdx(filter: { fields: { collection: { eq: "main" } } }) {
+          nodes {
+            frontmatter {
+              title
+              order
+            }
+            fields {
+              slug
+            }
+          }
+        }
+        components: allMdx(
+          filter: { fields: { collection: { eq: "components" } } }
+        ) {
+          nodes {
+            frontmatter {
+              title
+            }
+            fields {
+              slug
+            }
+          }
+        }
+        utilities: allMdx(
+          filter: { fields: { collection: { eq: "utilities" } } }
+        ) {
+          nodes {
+            frontmatter {
+              title
+            }
+            fields {
+              slug
+            }
+          }
+        }
+      }
+    `,
+  )
+}
+
+const MainLinks = () => {
+  const { main } = useLinksQuery()
+  const nodes = sortByOrder(main.nodes)
+
+  return nodes.map(({ frontmatter: { title }, fields: { slug } }) => (
+    <TopNavLink key={title} href={`/${slug}`}>
+      {title}
+    </TopNavLink>
+  ))
+}
+
+const ComponentLinks = () => {
+  const { components } = useLinksQuery()
+  const nodes = sortByTitle(components.nodes)
+
+  return nodes.map(({ frontmatter: { title }, fields: { slug } }) => (
+    <ComponentLink key={title} href={`/${slug}`}>
+      {upperFirst(camelCase(title))}
```

Comment:
This uses `frontmatter.title` to make sure all links are in the `PascalCase` format.